go/vt/sqlparser/sql.y

```diff
@@ -1529,9 +1542,9 @@ alter_table_statement:
   {
     $$ = &DDL{Action: AlterStr, Table: $4, IndexSpec: &IndexSpec{Action: CreateStr, ToName: $7,  Using: $8, Columns: $10, Options: $12}}
   }
-| ALTER ignore_opt TABLE table_name ADD constraint_symbol_opt constraint index_or_key sql_id using_opt '(' index_column_list ')' index_option_list_opt
+| ALTER ignore_opt TABLE table_name ADD constraint_symbol_opt key_type index_or_key_opt name_opt using_opt '(' index_column_list ')' index_option_list_opt
```

Comment:
Good catch on the `index_or_key_opt` here. Just re-checked the MySQL docs and that is indeed optional when a `key_type` is present. I definitely missed that, `{` and `[` look so similar without an IDE at times.